### ISSUE TITLE
List of directories to not watch for changes when serving locally

### ISSUE THREAD

**Issue by bender2k14** [CONNECT-platform/codedoc#52]
When locally serving my site, the page is reloading too much.  Looking at the terminal, it is because `codedoc serve` is issuing a reload when files in the `.git` directory change.
```
# change in C:\Code\Blogs\tyson-williams-blog\.git\FETCH_HEAD, issueing reload to client ...
# change in C:\Code\Blogs\tyson-williams-blog\.git\objects\tmp_object_git2_a17264, issueing reload to client ...
```

I think that directory should not be watched for changes for all CODEDOC users.

In addition to that, I have another directory that I would like to not be watched.  Sometimes I will include [actual code](https://github.com/bender2k14/tyson-williams-blog/tree/master/src/code/IdentityAsAs) in my repository to complement a [post](http://localhost:3000/2020-07-14_identity_as_as).  With Visual Studio open, I keep getting messages like this.
```
# change in C:\Code\Blogs\tyson-williams-blog\src\code\IdentityAsAs\.vs\IdentityAsAs\v16\.suo, issueing reload to client ...
```

But even when I am causing the change, I don't need the website reloaded.  When I make a change to source code file, a reload is issued.  This codes exists as a reference both for me to test the claims I make in my post and for others to also test those claims and play with the code.  No browser reloaded is needed.
```
# change in C:\Code\Blogs\tyson-williams-blog\src\code\IdentityAsAs\Project\Disambiguation.cs, issueing reload to client ...
```

As a side note, because my posts include the [utterances](https://utteranc.es/) script that calls the GitHub API on every page refresh, my IP is sometimes temporarily blocked by rate limiting.  It isn't a problem for me, but I would prefer to be more respectful of GitHub's resources.  Regardless of this reload issue, is there an easy why to disable the script when the site is locally hosted?  I can split this off into its own issue if that would be better.

**Comment by bender2k14** [comment]
Maybe this issue is related to https://github.com/CONNECT-platform/codedoc/issues/43#issuecomment-659455870
> it is notable that because of this, **CODEDOC** issues a reload to the client for ANY change in assets folder (excluding `.html` files). these reloads are of course not issued if a rebuild process is already ongoing, and as per my limited testing this seems to be working fine. however in different environments / configurations this might need further tweaking.

**Comment by loreanvictor** [comment]
it is. I guess I will actually go the other way around, only reloading in response to particular file extensions (`.jpg`, `.jpeg`, `.png`, `.gif`, `.svg`, `.webp`, `.woff`, `.ttf`, `.woff2`, `.otf`, `.eot`, `.css`. am I missing something?

**Comment by loreanvictor** [comment]
I suspect this should be a good start, since not picking up changes in resources isn't a drastic issue (its a manual page reload). Will restrict to this list and expand the list upon request/issue.

**Comment by bender2k14** [comment]
Ah, yes.  Whitelisting seems like a good approach to this problem.  Great idea.

**Comment by loreanvictor** [comment]
deployed via `0.2.10`

**Comment by bender2k14** [comment]
Works great.  No unneeded reloads but still reloads when I edit a `png` file.
